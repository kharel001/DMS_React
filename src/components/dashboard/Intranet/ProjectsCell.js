import React from 'react';
import IconButton from '@material-ui/core/IconButton';
// import Avatar from '@material-ui/core/es/Avatar/Avatar';
import LineIndicator from "./LineIndicator";

const ProjectsCell = ({data}) => {
  //originally img was Avatar, replaced it as it was creating difficulty during testing
  const {id, name, date, color, teamList, progressValue} = data;
  return (

    <tr
      tabIndex={-1}
      key={id}>

      <td className="max-width-100">
        <p className="text-truncate mb-0">{name}</p>
      </td>

      <td className="text-nowrap">{date}</td>
      <td>
        <LineIndicator width={progressValue / 4 + "%"} color={color} value={progressValue}/>
      </td>
      <td>
        <ul className="team-list">
          {teamList.map((team, index) =>
            <li key={team.id}>
              <img className="user-avatar" src={team.image}/> 
            </li>
          )}
        </ul>
      </td>
      <td>
        <IconButton className="text-light icon-btn p-1">
          <i className="zmdi zmdi-more-vert text-grey"/>
        </IconButton>
      </td>
    </tr>

  );
};

export default ProjectsCell;
