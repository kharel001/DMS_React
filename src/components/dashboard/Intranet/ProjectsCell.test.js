import React from 'react';
import ProjectsCell from './ProjectsCell';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const dummy_data={
    data:{
        id:1,
        name:"name1",
        date: "4 dec, 2020",
        color:"gray",
        teamList:[{id:1, image:"image1"}, {id:2, image:"image2"}],
        progressValue:67
    }
}
const setup = (props = {}) => {
    return shallow( < ProjectsCell {...props}  />
    )
}


it("ProjectsCell",()=>{
    const wrapper= setup(dummy_data);
    expect(wrapper).toMatchSnapshot();
})