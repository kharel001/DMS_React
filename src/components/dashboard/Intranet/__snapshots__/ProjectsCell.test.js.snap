// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsCell 1`] = `ShallowWrapper {}`;
